packages/phishing-controller/src/PhishingController.test.ts

```diff
@@ -2587,4 +2582,209 @@ describe('PhishingController', () => {
       expect(scope.isDone()).toBe(true);
     });
   });
+
+  describe('hotlistLastSuccessTimestamp handling', () => {
+    it('should update hotlistLastSuccessTimestamp only on successful response', async () => {
+      // Setup initial state
+      const controller = getPhishingController({
+        state: {
+          phishingLists: [
+            {
+              allowlist: [],
+              blocklist: ['185.149.120.83', '193.3.19.52', '209.159.154.156'],
+              c2DomainBlocklist: [],
+              fuzzylist: [],
+              tolerance: 0,
+              lastUpdated: 1742310773, // Just before the update
+              name: ListNames.MetaMask,
+              version: 0,
+            },
+          ],
+          hotlistLastSuccessTimestamp: 1742310773, // Previous successful timestamp
+        },
+      });
+
+      // Mock a successful response with lastFetchedAt at the root level similar to the example
+      nock(PHISHING_CONFIG_BASE_URL)
+        .get(`${METAMASK_HOTLIST_DIFF_FILE}/1742310773`)
+        .reply(200, {
+          data: [
+            {
+              isRemoval: true,
+              url: '185.149.120.83',
+              timestamp: 1742310774,
+              targetList: 'eth_phishing_detect_config.blocklist',
+            },
+            {
+              isRemoval: true,
+              url: '193.3.19.52',
+              timestamp: 1742310774,
+              targetList: 'eth_phishing_detect_config.blocklist',
+            },
+            {
+              isRemoval: true,
+              url: '209.159.154.156',
+              timestamp: 1742310774,
+              targetList: 'eth_phishing_detect_config.blocklist',
+            },
+          ],
+          lastFetchedAt: 1744213500,
+        });
+
+      await controller.updateHotlist();
+
+      // Verify the timestamp was updated
+      expect(controller.state.hotlistLastSuccessTimestamp).toBe(1744213500);
+
+      // We verify only the timestamp was updated, not checking blocklist
+      // since it depends on the implementation details of applyDiffs
+
+      // Mock a successful response WITHOUT lastFetchedAt
+      nock(PHISHING_CONFIG_BASE_URL)
+        .get(`${METAMASK_HOTLIST_DIFF_FILE}/1742310774`)
+        .reply(200, {
+          data: [],
+          // No lastFetchedAt property
+        });
+
+      await controller.updateHotlist();
+
+      // Verify it kept the previous value
+      expect(controller.state.hotlistLastSuccessTimestamp).toBe(1744213500);
+
+      // Mock a failed response
+      nock(PHISHING_CONFIG_BASE_URL)
+        .get(`${METAMASK_HOTLIST_DIFF_FILE}/1742310774`)
+        .reply(500);
+
+      await controller.updateHotlist();
+
+      // Verify the timestamp wasn't changed
+      expect(controller.state.hotlistLastSuccessTimestamp).toBe(1744213500);
+    });
```

Comment:
Here it seems that we are actually doing 3 tests:
* mock successfull response with lastFetchedAt
* mock successfull response without lastFetchedAt
* mock failed response

Each of those should be its own test (an `it` descriptor for each case). You can read more about testing best practices in the contributor docs:
* Overall best practices: https://github.com/nock/nock?tab=readme-ov-file#isdone
* Unit test best practices: https://github.com/MetaMask/contributor-docs/blob/main/docs/testing/unit-testing.md